command/agent/agent.go

```diff
@@ -119,7 +113,7 @@ func (c *cmd) startupJoin(agent *agent.Agent, cfg *config.RuntimeConfig) error {
 		return nil
 	}
 
-	c.UI.Output("Joining cluster...")
+	c.logger.Info("Joining cluster")
```

Comment:
This change to use a logger (and mentioned in the commit message). Using the logger for this seems much better, that way the output is not lost when json logging is enabled. 

Same below.